internal/analysis/hover_test.go

```diff
@@ -35,6 +35,54 @@ send $amt (
 	assert.NotNil(t, resolved)
 }
 
+func TestHoverOnSaveMonetaryVar(t *testing.T) {
+	input := `vars { monetary $amt }
+
+save $amt from @acc
+`
+
+	rng := parser.RangeOfIndexed(input, "$amt", 1)
+
+	program := parser.Parse(input).Value
+	hover := analysis.HoverOn(program, rng.Start)
+	assert.NotNil(t, hover)
```

Comment:
You should use require.NotNil instead of assert.NotNil.
If the assumtion fail, the code will continue, and trigger a panic later.

Same almost everywhere.